### PR TITLE
Initial Enlistment of Adobe Unified Extensibility Platform Integration.

### PR DESCRIPTION
**Pull request by krisnye**
This PR contains our initial packages that integrate UXP support into React Spectrum.

There is a mono-package at `packages/@adobe/react-spectrum-uxp` which is and always will be API compatible with `packages/@adobe/react-spectrum`.

The rest of the UXP specific packages will live within the `packages/@react-spectrum-uxp` folder.

This initial enlistment is only intended to start establishing best practices and get us started on the right path. We will be improving our process and adding more components before we get close to publishing any of these new packages.

See the RFC for more details: https://github.com/adobe/react-spectrum/blob/main/rfcs/2021-v3-uxp-integration.md

## ✅ Pull Request Checklist:

- [n/a] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [n/a] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [x] Filled out test instructions.
- [x] Updated documentation (if it already exists for this component).
- [n/a] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

Testing this requires a UXP environment that can run plugins. The `packages/@react-spectrum-uxp/storybook` module builds a custom UXP storybook plugin. To test this, you must build or watch that project and then run the plugin in the dist directory within the UXP demo application or another UXP platform.

## 🧢 Your Project:

Adobe, Unified Extensibility Platform Team.

### CODE REVIEW

File: packages/@react-spectrum-uxp/parcel-resolver-alias-fix/AliasFixResolver.js
Comment (snowystinger) [inline packages/@react-spectrum-uxp/parcel-resolver-alias-fix/AliasFixResolver.js:32]: There's an issue for upgrading parcel, do you think you could add a link into that issue or tag yourself in it so that you're updated when we do that?

Comment (snowystinger) [reply packages/@react-spectrum-uxp/parcel-resolver-alias-fix/AliasFixResolver.js:32]: https://github.com/adobe/react-spectrum/issues/1743

File: packages/@react-spectrum-uxp/utils/package.json
Comment (krisnye) [reply packages/@react-spectrum-uxp/utils/package.json:27]: This package was cloned from @react-spectrum/utils so there may be some vestigial imports. I'll try to remove them all.

File: packages/@react-spectrum-uxp/utils/package.json
Comment (krisnye) [reply packages/@react-spectrum-uxp/utils/package.json:26]: removed.

File: packages/@adobe/react-spectrum-uxp/README.md
Comment (krisnye) [reply packages/@adobe/react-spectrum-uxp/README.md:7]: I've added a link to the external UXP documentation. I'm not sure what to add regarding setting up this repo because it compiles and is used just like every other repo within this monorepo. This specific repo is homologous with @adobe/react-spectrum. I'll add more specific setup documentation in the @react-spectrum-uxp/storybook repository.

File: packages/@adobe/react-spectrum-uxp/README.md
Comment (krisnye) [reply packages/@adobe/react-spectrum-uxp/README.md:3]: Changed.

File: packages/@adobe/react-spectrum-uxp/package.json
Comment (krisnye) [reply packages/@adobe/react-spectrum-uxp/package.json:3]: done

File: packages/@react-spectrum-uxp/parcel-resolver-alias-fix/AliasFixResolver.js
Comment (krisnye) [reply packages/@react-spectrum-uxp/parcel-resolver-alias-fix/AliasFixResolver.js:32]: Tagged myself.

File: packages/@react-spectrum-uxp/storybook/README.md
Comment (snowystinger) [inline packages/@react-spectrum-uxp/storybook/README.md:13]: can it be changed to a range so it won't force the entire repo to resolve to 16.13.1?
for instance, if we tried to change to 17 (which we plan on having that be the default soon), then this would cause a problem no?

File: packages/@react-spectrum-uxp/storybook/package.json
Comment (snowystinger) [inline packages/@react-spectrum-uxp/storybook/package.json:8]: so to run uxp storybook, you'd expect a user to be at the root and use `lerna run --scope rs3uxp-plugin watch:parcel`? or do you expect them to be in this directory? or how does this work?

File: packages/@react-spectrum-uxp/storybook/package.json
Comment (snowystinger) [inline packages/@react-spectrum-uxp/storybook/package.json:11]: this seems very dependent on what you've done to your local system, can we generalize this a bit, maybe allow the path to be passed in? or instructions to add it to the PATH?
also, it's only going to work for Mac developers, is that alright?

File: packages/@react-spectrum-uxp/storybook/package.json
Comment (snowystinger) [reply packages/@react-spectrum-uxp/storybook/package.json:8]: you don't need to rm -rf the parcel-cache, just running parcel with the flag `--no-cache` that you have already should be fine
did you run into an issue with that?

File: packages/@react-spectrum-uxp/storybook/README.md
Comment (snowystinger) [reply packages/@react-spectrum-uxp/storybook/README.md:13]: you can find the range we use in other components https://github.com/adobe/react-spectrum/blob/main/packages/%40react-spectrum/button/package.json#L51

File: packages/@react-spectrum-uxp/storybook/package.json
Comment (krisnye) [reply packages/@react-spectrum-uxp/storybook/package.json:8]: I've run into a million issues with the parcel cache, so I always delete it as a standard parcel ritual. I guess --no-cache will do the trick though.

Concerning a user running storybook, currently, I run it from that directory. To start with, only the RS3-UXP developers will be running the storybook. As we get farther along, we will improve the documentation, improve the storybook UI and eventually we will provide instructions for external developers to be able to run the storybook as a plugin within XD (or other UXP plugin compatible product).

I know there is a ton of work to do before we reach that point. Initially I'm just trying to get the basic structure right so that I can open up RS3+UXP development to the main team that will be implementing the components.

File: packages/@react-spectrum-uxp/storybook/package.json
Comment (krisnye) [reply packages/@react-spectrum-uxp/storybook/package.json:11]: Not that specific to my system, but it is mac specific. I will change that to uxp-mac and later we can make a uxp-win.  This just requires torq-native to be installed as a peer directory to the monorepo. The simplicity of the script also makes it immediately clear to anyone capable of installing torq-native that they need to run the demo app (wherever theirs is located) and they can make their own simple shell script as needed.

I will definitely documenting the full setup so that anyone at Adobe can in principle set this up, but I expect that not many will want to setup torq-native (UXP) from scratch just for this.

Longer term, my goal is to provide a much easier way for RS3 developers to test components against UXP... either via a XD plugin or a standalone executable... possibly a UXP Browser (which I've been pushing for).

File: packages/@react-spectrum-uxp/storybook/README.md
Comment (krisnye) [reply packages/@react-spectrum-uxp/storybook/README.md:13]: Done.

**Review by snowystinger — DISMISSED** [review]

**Comment by snowystinger** [comment]
approved, assuming lint issue will be worked out

File: scripts/lint-packages.js
Comment (dannify) [inline scripts/lint-packages.js:21]: Why was this updated? I would prefer not to have this change for the whole repo.

Comment (krisnye) [reply scripts/lint-packages.js]: To avoid linting the parcel-resolver-alias-fix repo. Instead it looks like I can just move it to `packages/dev` in order to avoid linting.

**Review by snowystinger — APPROVED** [review]
Looks good, I checked docs build and that seems to be working. I haven't tried building a release of RSP, but that'd be another thing we should try. Only because the parcel resolver alias was touched.
I think the lint-packages change is ok now

**Comment by dannify** [comment]
This project is on hold for now. Can be reopened at a later date.

**Pull request closed without merging**
